plugin-dev/Source/Sentry/Private/SentrySubsystem.cpp

```diff
@@ -66,9 +71,9 @@ void USentrySubsystem::Initialize()
 		return;
 	}
 
-	if(!IsCurrentBuildConfigurationEnabled() || !IsCurrentBuildTargetEnabled())
+	if(!IsCurrentBuildConfigurationEnabled() || !IsCurrentBuildTargetEnabled() || !IsCurrentPlatformEnabled() || !IsPromotedBuild())
```

Comment:
How about checking whether the build is promoted separately? I believe the `IsPromotedBuild` method name doesn't reflect what it's doing accurately so maybe we should consider replacing it with something more explicit:
```cpp
if(Settings->EnableForPromotedBuildsOnly && !FApp::GetEngineIsPromotedBuild())
{
    UE_LOG(LogSentrySdk, Warning, TEXT("Sentry initialization skipped since event capturing is disabled for the non-promoted builds in plugin settings."));
    return;
}
```